spring-transaction/src/main/kotlin/org/jetbrains/exposed/spring/SpringTransactionManager.kt

```diff
@@ -107,6 +107,10 @@ class SpringTransactionManager(
             readOnly = definition.isReadOnly,
             outerTransaction = currentTransactionManager.currentOrNull()
         ).apply {
+            if (definition.timeout != TransactionDefinition.TIMEOUT_DEFAULT) {
+                queryTimeout = definition.timeout
+            }
```

Comment:
DefaultTimeout has a value of -1. If it is not this value, the user has set a new value, so I added this condition.